array/missing_ranges.py

```diff
@@ -1,27 +1,42 @@
-## find missing ranges between low and high in the given array.
-# ex) [3, 5] lo=1 hi=10 => answer: [1->2, 4, 6->10]
+"""
+Find missing ranges between low and high in the given array.
+Ex) [3, 5] lo=1 hi=10 => answer: [1->2, 4, 6->10]
+"""
+import unittest
+
+
+def missing_ranges(arr, lo, hi):
 
-def missing_ranges(nums, lo, hi):
     res = []
     start = lo
-    for num in nums:
-        if num < start:
-            continue
-        if num == start:
+
+    for n in arr:
```

Comment:
Test your changes for the case:  
```arr := [3,4]```  
and 
``` lo := 1 ```  
``` hi := 5 ```  
The orginal code gets  
```['1->2', '5']```  
Your code should return ```[[1,2],[2,3],[3,5]]```

Reply:
Thanks for the review.
Yeah, I changed that to the `list of tuples` version, I feel denoting ranges in `tuples` is better than `str`.
And, for yours above example, the code should return `[(1, 3), (5, 5)]` (and it does).
The range basically means all numbers included in range `(l, r)` from `l` to `r`.  
I could have let `5` be just `5` rather than `(5, 5)`, but I think since the output is in form of tuples, `(5, 5)` is consistent with it.

Reply:
Ok, I made a fail.